File: library/src/main/java/com/uploadcare/android/library/data/GroupPageData.java
package com.uploadcare.android.library.data;

import java.net.URI;
import java.util.List;

public class GroupPageData implements PageData<GroupData> {

    public URI next;

    public URI previous;

    public int total;

    public int perPage;

    public List<GroupData> results;

    public List<GroupData> getResults() {
        return results;
    }

    public boolean hasMore() {
        return next != null;
    }

    public URI getNext() {
        return next;
    }

}

File: library/src/main/java/com/uploadcare/android/library/data/GroupPageData.kt
package com.uploadcare.android.library.data

import com.squareup.moshi.Json
import com.uploadcare.android.library.api.UploadcareGroup
import java.net.URI

data class GroupPageData(val next: URI? = null,
                         val previous: URI? = null,
                         val total: Int,
                         @Json(name = "per_page") val perPage: Int,
                         val results: List<UploadcareGroup>) : PageData<UploadcareGroup> {

    override fun getResultsData() = results

    override fun hasMore() = next != null

    override fun getNextURI() = next
}